### PR TITLE
ns1: enable autoDNSSEC capability

### LINKED ISSUE #1444: NS1: Enable AUTODNSSEC capability
NS1 provides a way to enable/disable managed DNSSEC through their [DNSSEC API](https://ns1.com/api/#post-enable-dnssec-for-a-zone), with the only caveat being that DNSSEC capability has to be enabled across the account first via support.

The API part of the above process seems to map directly to the AUTODNSSEC capability.

Investigate whether that's actually the case for the ns1 provider and if yes, whether we can handle DNSSEC being disabled on the account level.

### PR DESCRIPTION
**Pull request by costasd**
NS1's API has support for enabling or disabling DNSSEC on zones under
DNSSEC-enabled accounts.

This commit enables support for detecting a zone's DNSSEC state and,
given the autoDNSSEC setting, pushing the right domain corrections
to our corrections stack.

Fixes #1444

### CODE REVIEW

File: providers/ns1/ns1Provider.go [hunk 1]
```diff
@@ -88,6 +89,52 @@ func (n *nsone) GetZoneRecords(domain string) (models.Records, error) {
 	}
 	return found, nil
 }
+// Get DNSSEC status for zone. Returns true for enabled, false for disabled
```
Comment (tlimoncelli) [inline providers/ns1/ns1Provider.go:92]: Go style: Start the comment with the name of the function.
```suggestion
// GetZoneDNSSEC gets the DNSSEC status for zone. Returns true for enabled, false for disabled
```

File: providers/ns1/ns1Provider.go [hunk 2]
```diff
@@ -88,6 +89,52 @@ func (n *nsone) GetZoneRecords(domain string) (models.Records, error) {
...
+
+	// any other errors not expected, let's surface them
+	if err != nil {
+		return false, err
+	}
+
+	// no errors returned, we assume DNSSEC is enabled
+	return true, nil
+}
+// Create DNSSEC zone corrections based on current state and preference
```
Comment (tlimoncelli) [inline providers/ns1/ns1Provider.go:113]:
```suggestion
// getDomainCorrectionsDNSSEC creates DNSSEC zone corrections based on current state and preference
```

File: providers/ns1/ns1Provider.go [hunk 3]
```diff
@@ -88,6 +89,52 @@ func (n *nsone) GetZoneRecords(domain string) (models.Records, error) {
...
+	}
+
+	// any other errors not expected, let's surface them
+	if err != nil {
+		return false, err
+	}
+
+	// no errors returned, we assume DNSSEC is enabled
+	return true, nil
+}
```
Comment (tlimoncelli) [inline providers/ns1/ns1Provider.go:112]: Add a blank line between each function unless there is a specific reason not to.

File: providers/ns1/ns1Provider.go [hunk 4]
```diff
@@ -158,6 +211,21 @@ func (n *nsone) modify(recs models.Records, domain string) error {
 	return err
 }
 
+// Configures DNSSEC for a zone. Set 'enabled' to true to enable, false to disable.
```
Comment (tlimoncelli) [inline providers/ns1/ns1Provider.go:214]:
```suggestion
// configureDNSSEC configures DNSSEC for a zone. Set 'enabled' to true to enable, false to disable.
```

File: providers/ns1/ns1Provider.go [hunk 5]
```diff
@@ -158,6 +211,21 @@ func (n *nsone) modify(recs models.Records, domain string) error {
...
+// There's a cornercase, in which DNSSEC is globally disabled for the account.
+// In that situation, enabling DNSSEC will always fail with:
+//   #1: ENABLE DNSSEC
+//   FAILURE! POST https://api.nsone.net/v1/zones/example.com: 400 DNSSEC support is not enabled for this account. Please contact support@ns1.com to enable it
+//
+// Unfortunately this is not detectable otherwise, so given that we have a nice error message, we just let this through.
+//
+func (n *nsone) configureDNSSEC(domain string, enabled bool) error {
+	z, _, err := n.Zones.Get(domain)
```
Comment (tlimoncelli) [inline providers/ns1/ns1Provider.go:223]: Check `err`.  (otherwise, it is overwritten by line 225)

**Review by tlimoncelli — COMMENTED** [review]
Excellent work so far!  Just a few minor suggestions.

File: providers/ns1/ns1Provider.go (hunk 1 above)
Comment (costasd) [reply providers/ns1/ns1Provider.go:92]: ah yeah, always forget these. Thanks!

File: providers/ns1/ns1Provider.go (hunk 2 above)
Comment (costasd) [reply providers/ns1/ns1Provider.go:113]: Ditto.

File: providers/ns1/ns1Provider.go (hunk 5 above)
Comment (costasd) [reply providers/ns1/ns1Provider.go:223]: good catch!

**Review by tlimoncelli — APPROVED** [review]

**Pull request merged**